File: coding/03rd January 2025/Reverse Bits.java
//Reverse Bits
public class Solution 
{
    // you need treat n as an unsigned value
    public int reverseBits(int n) 
    {
        int res = 0;
        for(int i = 0; i < 32; i++)
        {
            res = res << 1; //Shift left
            res = res + (n & 1);
            n = n >> 1; //right shift
        }
        return res;
    }
}

File: Coding in Python/16th February 2026/Reverse Bits.py
#Reverse Bits
class Solution:
    def reverseBits(self, n: int) -> int:
        binary = bin(n)[2:]
        binary = binary.zfill(32)
        binary = binary[::-1]
        decimal = 0
        for digit in binary:
            decimal = decimal * 2 + int(digit)
        return decimal
